app/manager-frontend/src/api/datasource.js

```diff
@@ -0,0 +1,26 @@
+import { useBackendApi } from "contexts/BackendApi";
+
+import useSWR from "swr";
+
+export const fetchKeys = {
+  dataSourceList: "DataSources",
+};
+
+export const getDataSourceApi = ({ api }) => ({
+  delete: ({ values, id }) =>
+    api.delete(`DataSources/${id}`, {
+      json: values,
+    }),
+});
+
+export const useDataSourceList = () => {
+  const { apiFetcher } = useBackendApi();
+  return useSWR(
+    fetchKeys.dataSourceList,
+    async (url) => {
+      const data = await apiFetcher(url);
+      return data;
+    },
+    { suspense: true }
+  );
+};
```

Comment:
```suggestion
export const useDataSourceList = () => {
  const { apiFetcher } = useBackendApi();
  return useSWR(
    fetchKeys.dataSourceList,
    apiFetcher,
    { suspense: true }
  );
};
```
if all you're doing is calling apiFetcher, can't you just pass apiFetcher directly, instead of wrapping another unnecessary closure around it? it's already an async function that takes a url and returns data.